### ISSUE TITLE
Runtime Exception Error Message During Test

### ISSUE THREAD

**Issue by iroja004** [printix/Power-BI#8]
Hello, hopefully someone can help me out a bit, when I run the test i receive

Failed
 
Exception: System.Management.Automation.RuntimeException: You cannot call a method on a null-valued expression.
   at System.Management.Automation.ExceptionHandlingOps.CheckActionPreference(FunctionContext funcContext, Exception exception)
   at System.Management.Automation.Interpreter.ActionCallInstruction`2.Run(InterpretedFrame frame)
   at System.Management.Automation.Interpreter.EnterTryCatchFinallyInstruction.Run(InterpretedFrame frame)
   at System.Management.Automation.Interpreter.EnterTryCatchFinallyInstruction.Run(InterpretedFrame frame)
   at System.Management.Automation.Interpreter.Interpreter.Run(InterpretedFrame frame)
   at System.Management.Automation.Interpreter.LightLambda.RunVoid1[T0](T0 arg0)
   at System.Management.Automation.PSScriptCmdlet.RunClause(Action`1 clause, Object dollarUnderbar, Object inputToProcess)
   at System.Management.Automation.CommandProcessorBase.Complete() 
Activity: New-PrintixDataExtract 
Error Category: InvalidOperation 
Error Reason: RuntimeException ( 
Exception: System.Management.Automation.RuntimeException: You cannot call a method on a null-valued expression.
   at System.Management.Automation.ExceptionHandlingOps.CheckActionPreference(FunctionContext funcContext, Exception exception)
   at System.Management.Automation.Interpreter.ActionCallInstruction`2.Run(InterpretedFrame frame)
   at System.Management.Automation.Interpreter.EnterTryCatchFinallyInstruction.Run(InterpretedFrame frame)
   at System.Management.Automation.Interpreter.EnterTryCatchFinallyInstruction.Run(InterpretedFrame frame)
   at System.Management.Automation.Interpreter.Interpreter.Run(InterpretedFrame frame)
   at System.Management.Automation.Interpreter.LightLambda.RunVoid1[T0](T0 arg0)
   at System.Management.Automation.PSScriptCmdlet.RunClause(Action`1 clause, Object dollarUnderbar, Object inputToProcess)
   at System.Management.Automation.CommandProcessorBase.Complete() 
Activity: New-PrintixDataExtract 
Error Category: InvalidOperation 
Error Reason: RuntimeException)

This happens when it requests the actual data extract. The last message I receive is

POST https://auth.printix.net/oauth/token with -1-byte payload

Any assistance would be greatly appreciated.

Thanks,
Ivan

**Comment by MortenRei** [comment]
Hello @iroja004 

Are you running the Get-PrintixDirectCustomerDataExtract.ps1 script? Could you please change "$VerbosePreference = 'SilentlyContinue'" to "$VerbosePreference = 'Continue' and repost the output.

**Comment by MortenRei** [comment]
Thanks. Need to improve the output on those kind of errors.

So, it fails on getting a token from the printix API. That means there's something wrong with the credentials provided. Could you please verify the following;
- TenantDomain (line 50 in Get-PrintixDirectCustomerDataExtract.ps1)
- Check that the username (Printix API ClientID ) and password (Secret) you stored in the credential asset in step 1.2 (https://github.com/printix/Power-BI/blob/master/Documentation/ReadmeForDirectCustomers.md) is correct, and that you provide the correct credential name in the  $ClientCredentialsName variable.

**Comment by iroja004** [comment]
Hi Morten,

Thank you for the input. I've verified that the information in the script matches with the information provided by Printix. It seems to be my tenant ID. I get an error message when I try a simple call

I've reached back out to Printix to see if they can check on any possible issues with my Printix tenant.

I'll keep you posted on the progress.

Thanks!

**Comment by iroja004** [comment]
Hi, Morten,

I've corrected the information being used in the credentials asset. I had the something different being used for the username (not the ClientID). In addition, I've changed the tenantID to the one related to our TenantDomain (verified by Printix to be correct now). This gets me past the POST call, but it's still returning with null. Below is the updated log:

22:40:16 - Successfuly connected to Azure with Service Principal.
22:40:16 - Getting AutomationPSCredential [PrintixAPICredentials].
22:40:17 -   Requesting data extract. This might take some time...
22:40:18 - Authentication headers expired. Getting new authentication headers
22:40:18 - Setting Printix HTTP Headers
22:40:18 - Getting Printix Authorization Token
22:40:18 - Setting body
22:40:18 - Getting Authentication headers
POST https://auth.printix.net/oauth/token with -1-byte payload
received -1-byte response of content type application/json;charset=ISO-8859-1
22:40:19 - Setting script expire time
22:40:19 - Return Authentication headers


Ivan
